container/pull.bzl

```diff
@@ -31,6 +31,24 @@ def python(repository_ctx):
     fail("rules_docker requires a python interpreter installed. " +
          "Please set BAZEL_PYTHON, or put it on your path.")
 
+_container_pull_attrs = {
+    "registry": attr.string(mandatory = True),
+    "repository": attr.string(mandatory = True),
+    "digest": attr.string(),
+    "tag": attr.string(default = "latest"),
+    "os": attr.string(default = "linux"),
+    "os_version": attr.string(),
+    "os_features": attr.string_list(),
+    "architecture": attr.string(default = "amd64"),
+    "cpu_variant": attr.string(),
+    "platform_features": attr.string_list(),
```

Comment:
I'm not particularly happy about the name of this attribute, but `features` is a reserved attribute.

The [docker manifest list spec](https://docs.docker.com/registry/spec/manifest-v2-2/#manifest-list-field-descriptions) refers to this as CPU features, but the [OCI image index spec](https://github.com/opencontainers/image-spec/blob/master/image-index.md) just says it's a reserved field.

Reply:
I haven't found any images that actually use `features` yet.